### DISCUSSION TITLE
Import Error for pytests

### DISCUSSION THREAD

**Discussion by sravanAruru** [pytest-dev/pytest#12010] [Q&A]
Hi Team ,

While running pytests am getting below error 

(/Users/<username>/Library/Caches/pypoetry/virtualenvs/abc-ei44LHuz-py3.8/lib/python3.8/site-packages/markupsafe/
../../../../Library/Caches/pypoetry/virtualenvs/abc-ei44LHuz-py3.8/lib/python3.8/site-packages/jinja2/filters.py:13: in <module>
    from markupsafe import soft_unicode
**E   ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/Users/usrname/Library/Caches/pypoetry/virtualenvs/abc-ei44LHuz-py3.8/lib/python3.8/site-packages/markupsafe/__init__.py)**

And the below are the package version am using

├── python-dateutil >=2.7.0
│   └── six >=1.5 
└── types-python-dateutil >=2.8.10
aws-sam-cli 1.105.0 AWS SAM CLI is a CLI tool for local development and testing of Serverless applications
├── aws-lambda-builders 1.43.0
│   ├── setuptools * 
│   └── wheel * 
├── aws-sam-translator 1.82.0
│   ├── boto3 >=1.19.5,<2.dev0 
│   │   ├── botocore >=1.34.43,<1.35.0 
│   │   │   ├── jmespath >=0.7.1,<2.0.0 
│   │   │   ├── python-dateutil >=2.1,<3.0.0 
│   │   │   │   └── six >=1.5 
│   │   │   ├── urllib3 >=1.25.4,<1.27 
│   │   │   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
│   │   ├── jmespath >=0.7.1,<2.0.0 (circular dependency aborted here)
│   │   └── s3transfer >=0.10.0,<0.11.0 
│   │       └── botocore >=1.33.2,<2.0a.0 (circular dependency aborted here)
│   ├── jsonschema >=3.2,<5 
│   │   ├── attrs >=22.2.0 
│   │   ├── importlib-resources >=1.4.0 
│   │   │   └── zipp >=3.1.0 
│   │   ├── jsonschema-specifications >=2023.03.6 
│   │   │   ├── importlib-resources >=1.4.0 (circular dependency aborted here)
│   │   │   └── referencing >=0.31.0 
│   │   │       ├── attrs >=22.2.0 (circular dependency aborted here)
│   │   │       └── rpds-py >=0.7.0 
│   │   ├── pkgutil-resolve-name >=1.3.10 
│   │   ├── referencing >=0.28.4 (circular dependency aborted here)
│   │   └── rpds-py >=0.7.1 (circular dependency aborted here)
│   ├── pydantic >=1.8,<3 
│   │   ├── annotated-types >=0.4.0 
│   │   │   └── typing-extensions >=4.0.0 
│   │   ├── pydantic-core 2.16.2 
│   │   │   └── typing-extensions >=4.6.0,<4.7.0 || >4.7.0 (circular dependency aborted here)
│   │   └── typing-extensions >=4.6.1 (circular dependency aborted here)
│   └── typing-extensions >=4.4,<5 (circular dependency aborted here)
├── boto3 >=1.26.109,<2
│   ├── botocore >=1.34.43,<1.35.0 
│   │   ├── jmespath >=0.7.1,<2.0.0 
│   │   ├── python-dateutil >=2.1,<3.0.0 
│   │   │   └── six >=1.5 
│   │   ├── urllib3 >=1.25.4,<1.27 
│   │   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
│   ├── jmespath >=0.7.1,<2.0.0 (circular dependency aborted here)
│   └── s3transfer >=0.10.0,<0.11.0 
│       └── botocore >=1.33.2,<2.0a.0 (circular dependency aborted here)
├── boto3-stubs 1.33.5
│   ├── botocore-stubs * 
│   │   ├── types-awscrt * 
│   │   └── typing-extensions >=4.1.0 
│   ├── mypy-boto3-apigateway >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-cloudformation >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-ecr >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-iam >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-kinesis >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-lambda >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-s3 >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-schemas >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-secretsmanager >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-signer >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-sqs >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-stepfunctions >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-sts >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── mypy-boto3-xray >=1.33.0,<1.34.0 
│   │   └── typing-extensions >=4.1.0 (circular dependency aborted here)
│   ├── types-s3transfer * 
│   └── typing-extensions >=4.1.0 (circular dependency aborted here)
├── cfn-lint >=0.83.4,<0.84.0
│   ├── aws-sam-translator >=1.82.0 
│   │   ├── boto3 >=1.19.5,<2.dev0 
│   │   │   ├── botocore >=1.34.43,<1.35.0 
│   │   │   │   ├── jmespath >=0.7.1,<2.0.0 
│   │   │   │   ├── python-dateutil >=2.1,<3.0.0 
│   │   │   │   │   └── six >=1.5 
│   │   │   │   ├── urllib3 >=1.25.4,<1.27 
│   │   │   │   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
│   │   │   ├── jmespath >=0.7.1,<2.0.0 (circular dependency aborted here)
│   │   │   └── s3transfer >=0.10.0,<0.11.0 
│   │   │       └── botocore >=1.33.2,<2.0a.0 (circular dependency aborted here)
│   │   ├── jsonschema >=3.2,<5 
│   │   │   ├── attrs >=22.2.0 
│   │   │   ├── importlib-resources >=1.4.0 
│   │   │   │   └── zipp >=3.1.0 
│   │   │   ├── jsonschema-specifications >=2023.03.6 
│   │   │   │   ├── importlib-resources >=1.4.0 (circular dependency aborted here)
│   │   │   │   └── referencing >=0.31.0 
│   │   │   │       ├── attrs >=22.2.0 (circular dependency aborted here)
│   │   │   │       └── rpds-py >=0.7.0 
│   │   │   ├── pkgutil-resolve-name >=1.3.10 
│   │   │   ├── referencing >=0.28.4 (circular dependency aborted here)
│   │   │   └── rpds-py >=0.7.1 (circular dependency aborted here)
│   │   ├── pydantic >=1.8,<3 
│   │   │   ├── annotated-types >=0.4.0 
│   │   │   │   └── typing-extensions >=4.0.0 
│   │   │   ├── pydantic-core 2.16.2 
│   │   │   │   └── typing-extensions >=4.6.0,<4.7.0 || >4.7.0 (circular dependency aborted here)
│   │   │   └── typing-extensions >=4.6.1 (circular dependency aborted here)
│   │   └── typing-extensions >=4.4,<5 (circular dependency aborted here)
│   ├── jschema-to-python >=1.2.3,<1.3.0 
│   │   ├── attrs * (circular dependency aborted here)
│   │   ├── jsonpickle * 
│   │   └── pbr * 
│   ├── jsonpatch * 
│   │   └── jsonpointer >=1.9 
│   ├── jsonschema >=3.0,<5 (circular dependency aborted here)
│   ├── junit-xml >=1.9,<2.0 
│   │   └── six * (circular dependency aborted here)
│   ├── networkx >=2.4,<4 
│   ├── pyyaml >5.4 
│   ├── regex >=2021.7.1 
│   ├── sarif-om >=1.0.4,<1.1.0 
│   │   ├── attrs * (circular dependency aborted here)
│   │   └── pbr * (circular dependency aborted here)
│   └── sympy >=1.0.0 
│       └── mpmath >=0.19 
├── chevron >=0.12,<1.0
├── click >=8.1,<9.0
│   └── colorama * 
├── cookiecutter >=2.5.0,<2.6.0
│   ├── arrow * 
│   │   ├── python-dateutil >=2.7.0 
│   │   │   └── six >=1.5 
│   │   └── types-python-dateutil >=2.8.10 
│   ├── binaryornot >=0.4.4 
│   │   └── chardet >=3.0.2 
│   ├── click >=7.0,<9.0.0 
│   │   └── colorama * 
│   ├── jinja2 >=2.7,<4.0.0 
│   │   └── markupsafe >=0.23 
│   ├── python-slugify >=4.0.0 
│   │   └── text-unidecode >=1.3 
│   ├── pyyaml >=5.3.1 
│   ├── requests >=2.23.0 
│   │   ├── certifi >=2017.4.17 
│   │   ├── charset-normalizer >=2,<4 
│   │   ├── idna >=2.5,<4 
│   │   └── urllib3 >=1.21.1,<3 
│   └── rich * 
│       ├── markdown-it-py >=2.2.0 
│       │   └── mdurl >=0.1,<1.0 
│       ├── pygments >=2.13.0,<3.0.0 
│       └── typing-extensions >=4.0.0,<5.0 
├── dateparser >=1.2,<2.0
│   ├── python-dateutil * 
│   │   └── six >=1.5 
│   ├── pytz * 
│   ├── regex <2019.02.19 || >2019.02.19,<2021.8.27 || >2021.8.27 
│   └── tzlocal * 
│       ├── backports-zoneinfo * 
│       └── tzdata * 
├── docker >=6.1.0,<6.2.0
│   ├── packaging >=14.0 
│   ├── pywin32 >=304 
│   ├── requests >=2.26.0 
│   │   ├── certifi >=2017.4.17 
│   │   ├── charset-normalizer >=2,<4 
│   │   ├── idna >=2.5,<4 
│   │   └── urllib3 >=1.21.1,<3 
│   ├── urllib3 >=1.26.0 (circular dependency aborted here)
│   └── websocket-client >=0.32.0 
├── flask <3.1
│   ├── click >=5.1 
│   │   └── colorama * 
│   ├── itsdangerous >=0.24 
│   ├── jinja2 >=2.10.1 
│   │   └── markupsafe >=0.23 
│   └── werkzeug >=0.15 
│       └── markupsafe >=2.1.1 (circular dependency aborted here)
├── jmespath >=1.0.1,<1.1.0
├── jsonschema <4.21
│   ├── attrs >=22.2.0 
│   ├── importlib-resources >=1.4.0 
│   │   └── zipp >=3.1.0 
│   ├── jsonschema-specifications >=2023.03.6 
│   │   ├── importlib-resources >=1.4.0 (circular dependency aborted here)
│   │   └── referencing >=0.31.0 
│   │       ├── attrs >=22.2.0 (circular dependency aborted here)
│   │       └── rpds-py >=0.7.0 
│   ├── pkgutil-resolve-name >=1.3.10 
│   ├── referencing >=0.28.4 (circular dependency aborted here)
│   └── rpds-py >=0.7.1 (circular dependency aborted here)
├── pyopenssl >=23.3.0,<23.4.0
│   └── cryptography >=41.0.5,<42 
│       └── cffi >=1.12 
│           └── pycparser * 
├── pyyaml >=6.0.1,<7.0
├── regex !=2021.10.8
├── requests >=2.31.0,<2.32.0
│   ├── certifi >=2017.4.17 
│   ├── charset-normalizer >=2,<4 
│   ├── idna >=2.5,<4 
│   └── urllib3 >=1.21.1,<3 
├── rich >=13.7.0,<13.8.0
│   ├── markdown-it-py >=2.2.0 
│   │   └── mdurl >=0.1,<1.0 
│   ├── pygments >=2.13.0,<3.0.0 
│   └── typing-extensions >=4.0.0,<5.0 
├── ruamel-yaml >=0.18.5,<0.19.0
│   └── ruamel-yaml-clib >=0.2.7 
├── tomlkit 0.12.3
├── typing-extensions >=4.4.0,<5
├── tzlocal 5.2
│   ├── backports-zoneinfo * 
│   └── tzdata * 
└── watchdog 3.0.0
aws-wsgi 0.2.7 WSGI adapter for AWS API Gateway/Lambda Proxy Integration
aws-xray-sdk 2.12.1 The AWS X-Ray SDK for Python (the SDK) enables Python developers to record and emit information from within their applications to the AWS X-Ray service.
├── botocore >=1.11.3
│   ├── jmespath >=0.7.1,<2.0.0 
│   ├── python-dateutil >=2.1,<3.0.0 
│   │   └── six >=1.5 
│   ├── urllib3 >=1.25.4,<1.27 
│   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
└── wrapt *
awscli 1.32.43 Universal Command Line Environment for AWS.
├── botocore 1.34.43
│   ├── jmespath >=0.7.1,<2.0.0 
│   ├── python-dateutil >=2.1,<3.0.0 
│   │   └── six >=1.5 
│   ├── urllib3 >=1.25.4,<1.27 
│   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
├── colorama >=0.2.5,<0.4.5
├── docutils >=0.10,<0.17
├── pyyaml >=3.10,<6.1
├── rsa >=3.1.2,<4.8
│   └── pyasn1 >=0.1.3 
└── s3transfer >=0.10.0,<0.11.0
    └── botocore >=1.33.2,<2.0a.0 
        ├── jmespath >=0.7.1,<2.0.0 
        ├── python-dateutil >=2.1,<3.0.0 
        │   └── six >=1.5 
        ├── urllib3 >=1.25.4,<1.27 
        └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
bandit 1.7.7 Security oriented static analyser for python code.
├── colorama >=0.3.9
├── pyyaml >=5.3.1
├── rich *
│   ├── markdown-it-py >=2.2.0 
│   │   └── mdurl >=0.1,<1.0 
│   ├── pygments >=2.13.0,<3.0.0 
│   └── typing-extensions >=4.0.0,<5.0 
└── stevedore >=1.20.0
    └── pbr >=2.0.0,<2.1.0 || >2.1.0 
black 19.10b0 The uncompromising code formatter.
├── appdirs *
├── attrs >=18.1.0
├── click >=6.5
│   └── colorama * 
├── pathspec >=0.6,<1
├── regex *
├── toml >=0.9.4
└── typed-ast >=1.4.0
cfn-lint 0.83.8 Checks CloudFormation templates for practices and behaviour that could potentially be improved
├── aws-sam-translator >=1.82.0
│   ├── boto3 >=1.19.5,<2.dev0 
│   │   ├── botocore >=1.34.43,<1.35.0 
│   │   │   ├── jmespath >=0.7.1,<2.0.0 
│   │   │   ├── python-dateutil >=2.1,<3.0.0 
│   │   │   │   └── six >=1.5 
│   │   │   ├── urllib3 >=1.25.4,<1.27 
│   │   │   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
│   │   ├── jmespath >=0.7.1,<2.0.0 (circular dependency aborted here)
│   │   └── s3transfer >=0.10.0,<0.11.0 
│   │       └── botocore >=1.33.2,<2.0a.0 (circular dependency aborted here)
│   ├── jsonschema >=3.2,<5 
│   │   ├── attrs >=22.2.0 
│   │   ├── importlib-resources >=1.4.0 
│   │   │   └── zipp >=3.1.0 
│   │   ├── jsonschema-specifications >=2023.03.6 
│   │   │   ├── importlib-resources >=1.4.0 (circular dependency aborted here)
│   │   │   └── referencing >=0.31.0 
│   │   │       ├── attrs >=22.2.0 (circular dependency aborted here)
│   │   │       └── rpds-py >=0.7.0 
│   │   ├── pkgutil-resolve-name >=1.3.10 
│   │   ├── referencing >=0.28.4 (circular dependency aborted here)
│   │   └── rpds-py >=0.7.1 (circular dependency aborted here)
│   ├── pydantic >=1.8,<3 
│   │   ├── annotated-types >=0.4.0 
│   │   │   └── typing-extensions >=4.0.0 
│   │   ├── pydantic-core 2.16.2 
│   │   │   └── typing-extensions >=4.6.0,<4.7.0 || >4.7.0 (circular dependency aborted here)
│   │   └── typing-extensions >=4.6.1 (circular dependency aborted here)
│   └── typing-extensions >=4.4,<5 (circular dependency aborted here)
├── jschema-to-python >=1.2.3,<1.3.0
│   ├── attrs * 
│   ├── jsonpickle * 
│   └── pbr * 
├── jsonpatch *
│   └── jsonpointer >=1.9 
├── jsonschema >=3.0,<5
│   ├── attrs >=22.2.0 
│   ├── importlib-resources >=1.4.0 
│   │   └── zipp >=3.1.0 
│   ├── jsonschema-specifications >=2023.03.6 
│   │   ├── importlib-resources >=1.4.0 (circular dependency aborted here)
│   │   └── referencing >=0.31.0 
│   │       ├── attrs >=22.2.0 (circular dependency aborted here)
│   │       └── rpds-py >=0.7.0 
│   ├── pkgutil-resolve-name >=1.3.10 
│   ├── referencing >=0.28.4 (circular dependency aborted here)
│   └── rpds-py >=0.7.1 (circular dependency aborted here)
├── junit-xml >=1.9,<2.0
│   └── six * 
├── networkx >=2.4,<4
├── pyyaml >5.4
├── regex >=2021.7.1
├── sarif-om >=1.0.4,<1.1.0
│   ├── attrs * 
│   └── pbr * 
└── sympy >=1.0.0
    └── mpmath >=0.19 
coverage 5.5 Code coverage measurement for Python
dacite 1.8.1 Simple creation of data classes from dictionaries.
defusedxml 0.6.0 XML bomb protection for Python stdlib modules
doit 0.32.0 doit - Automation Tool
├── cloudpickle *
├── macfsevents *
└── pyinotify *
faker 4.18.0 Faker is a Python package that generates fake data for you.
├── python-dateutil >=2.4
│   └── six >=1.5 
└── text-unidecode 1.3
flake8 3.9.2 the modular source code checker: pep8 pyflakes and co
├── mccabe >=0.6.0,<0.7.0
├── pycodestyle >=2.7.0,<2.8.0
└── pyflakes >=2.3.0,<2.4.0
flask-dotenv 0.1.2 The .env file support for Flask
└── flask *
    ├── click >=5.1 
    │   └── colorama * 
    ├── itsdangerous >=0.24 
    ├── jinja2 >=2.10.1 
    │   └── markupsafe >=0.23 
    └── werkzeug >=0.15 
        └── markupsafe >=2.1.1 (circular dependency aborted here)
flask-rebar 2.1.1 Flask-Rebar combines flask, marshmallow, and swagger for robust REST services.
├── flask >=1.0,<3
│   ├── click >=5.1 
│   │   └── colorama * 
│   ├── itsdangerous >=0.24 
│   ├── jinja2 >=2.10.1 
│   │   └── markupsafe >=0.23 
│   └── werkzeug >=0.15 
│       └── markupsafe >=2.1.1 (circular dependency aborted here)
└── marshmallow >=3.0,<4
    └── packaging >=17.0 
fleece 0.18.7 Wrap the lamb...da
├── boto3 >=1.0.0
│   ├── botocore >=1.34.43,<1.35.0 
│   │   ├── jmespath >=0.7.1,<2.0.0 
│   │   ├── python-dateutil >=2.1,<3.0.0 
│   │   │   └── six >=1.5 
│   │   ├── urllib3 >=1.25.4,<1.27 
│   │   └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
│   ├── jmespath >=0.7.1,<2.0.0 (circular dependency aborted here)
│   └── s3transfer >=0.10.0,<0.11.0 
│       └── botocore >=1.33.2,<2.0a.0 (circular dependency aborted here)
├── requests >=2.9.1
│   ├── certifi >=2017.4.17 
│   ├── charset-normalizer >=2,<4 
│   ├── idna >=2.5,<4 
│   └── urllib3 >=1.21.1,<3 
├── structlog >=15.3.0
│   └── six * 
└── wrapt >=1.10.10
jinja2 2.11.3 A very fast and expressive template engine.
└── markupsafe >=0.23
marshmallow-enum 1.5.1 Enum field for Marshmallow
└── marshmallow >=2.0.0
    └── packaging >=17.0 
mock 3.0.5 Rolling backport of unittest.mock for all Pythons
└── six *
mypy 0.761 Optional static typing for Python
├── mypy-extensions >=0.4.3,<0.5.0
├── typed-ast >=1.4.0,<1.5.0
└── typing-extensions >=3.7.4
paramiko 2.12.0 SSH2 protocol library
├── bcrypt >=3.1.3
├── cryptography >=2.5
│   └── cffi >=1.12 
│       └── pycparser * 
├── pynacl >=1.0.1
│   └── cffi >=1.4.1 
│       └── pycparser * 
└── six *
passwordgenerator 1.5.1 Passwords easy for humans, hard for computers
pdpyras 4.5.2 PagerDuty Python REST API Sessions
├── requests *
│   ├── certifi >=2017.4.17 
│   ├── charset-normalizer >=2,<4 
│   ├── idna >=2.5,<4 
│   └── urllib3 >=1.21.1,<3 
└── urllib3 *
pynamodb 5.5.1 A Pythonic Interface to DynamoDB
└── botocore >=1.12.54
    ├── jmespath >=0.7.1,<2.0.0 
    ├── python-dateutil >=2.1,<3.0.0 
    │   └── six >=1.5 
    ├── urllib3 >=1.25.4,<1.27 
    └── urllib3 >=1.25.4,<2.1 (circular dependency aborted here)
pyparsing 3.1.1 pyparsing module - Classes and methods to define and execute parsing grammars
pytest 5.4.3 pytest: simple powerful testing with Python
├── atomicwrites >=1.0
├── attrs >=17.4.0
├── colorama *
├── more-itertools >=4.0.0
├── packaging *
├── pluggy >=0.12,<1.0
├── py >=1.5.0
└── wcwidth *
pytest-env 0.6.2 py.test plugin that allows you to add environment variables.
└── pytest >=2.6.0
    ├── atomicwrites >=1.0 
    ├── attrs >=17.4.0 
    ├── colorama * 
    ├── more-itertools >=4.0.0 
    ├── packaging * 
    ├── pluggy >=0.12,<1.0 
    ├── py >=1.5.0 
    └── wcwidth * 
python-dotenv 0.10.5 Add .env support to your django/flask apps in development and deployments
pyvcloud 21.0.1 VMware vCloud Director Python SDK
├── humanfriendly >=4.8
│   └── pyreadline3 * 
├── lxml >=4.2.1
├── pygments >=2.2.0
├── pyyaml >=4.2b1
├── requests >=2.18.4
│   ├── certifi >=2017.4.17 
│   ├── charset-normalizer >=2,<4 
│   ├── idna >=2.5,<4 
│   └── urllib3 >=1.21.1,<3 
└── unittest-xml-reporting >=2.2.1
    └── lxml * 
pyvmomi 6.7.3 VMware vSphere Python SDK
├── requests >=2.3.0
│   ├── certifi >=2017.4.17 
│   ├── charset-normalizer >=2,<4 
│   ├── idna >=2.5,<4 
│   └── urllib3 >=1.21.1,<3 
└── six >=1.7.3
pyyaml 6.0.1 YAML parser and emitter for Python
structlog 19.2.0 Structured Logging for Python
└── six *
taskipy 1.12.2 tasks runner for python projects
├── colorama >=0.4.4,<0.5.0
├── mslex >=1.1.0,<2.0.0
├── psutil >=5.7.2,<6.0.0
└── tomli >=2.0.1,<3.0.0
toml 0.10.2 Python Library for Tom's Obvious, Minimal Language
urllib3 1.26.18 HTTP library with thread-safe connection pooling, file post, and more.

**Comment by RonnyPfannschmidt** [comment]
at first glance it looks like your jinja version is not compatible with the markupsafe version
pytest seems unrelated

**Reply by sravanAruru** [reply]
I tried different combinations of flaskrebar ,jinja and markupsafe any specific versions to be used to get this issue resolved ?

**Reply by RonnyPfannschmidt** [reply]
That's unclear,I'm currently not using that ecosystem, please check upstream
